### PR TITLE
Update JavaExecGeneratorTest.java

### LINKED ISSUE #3106: JavaExecGeneratorTest : Replace AssertJ's deprecated `asList()` DSL method with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`
### Component

JKube Kit

### Description
In the latest versions of Assertj `asList()` DSL method has been marked as deprecated:

https://github.com/eclipse-jkube/jkube/blob/94f2303221a4b85bde16a29874cb1803e3e61dcc/jkube-kit/generator/java-exec/src/test/java/org/eclipse/jkube/generator/javaexec/JavaExecGeneratorTest.java#L114-L117

users are advised to use `asInstanceOf(InstanceOfAssertFactories.list(type.class))` instead, we need to update `asList()` calls with this. 

### Expected Behavior
AssertJ's `asList()` is replaced with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`

### Acceptance Criteria
- [ ] `asList()` DSL method call is replaced with  `asInstanceOf(InstanceOfAssertFactories.list(type.class))` in all places

### PR DESCRIPTION
**Pull request by bhiwagade-rahul**
JavaExecGeneratorTest : Replace AssertJ's deprecated asList() DSL method with asInstanceOf(InstanceOfAssertFactories.list(type.class)) #3106

## Description

Fixes #3106

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3117 (_2024-05-30T16:10:13Z_)

:gear: [JKube E2E Tests (9305735492)](https://github.com/eclipse-jkube/ci/actions/runs/9305735492) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3117)](https://github.com/eclipse-jkube/ci/actions/runs/9305735492/job/25613371249)

File: jkube-kit/generator/java-exec/src/test/java/org/eclipse/jkube/generator/javaexec/JavaExecGeneratorTest.java [hunk 1]
```diff
@@ -111,9 +111,9 @@ void createAssemblyWithNoFatJarShouldAddDefaultFileSets() {
     // Then
     assertThat(result)
         .hasFieldOrPropertyWithValue("excludeFinalOutputArtifact", false)
-        .extracting(AssemblyConfiguration::getLayers).asList().hasSize(1)
+        .extracting(AssemblyConfiguration::getLayers).asInstanceOf(InstanceOfAssertFactories.list(Assembly.class)).hasSize(1)
         .first().asInstanceOf(InstanceOfAssertFactories.type(Assembly.class))
-        .extracting(Assembly::getFileSets).asList()
+        .extracting(Assembly::getFileSets).asInstanceOf(InstanceOfAssertFactories.list(Assembly.class))
```
Comment (rohanKanojia) [inline jkube-kit/generator/java-exec/src/test/java/org/eclipse/jkube/generator/javaexec/JavaExecGeneratorTest.java:116]: Shouldn't we be using AssemblyFileSet here?

Comment (bhiwagade-rahul) [reply jkube-kit/generator/java-exec/src/test/java/org/eclipse/jkube/generator/javaexec/JavaExecGeneratorTest.java]: Thanks @rohanKanojia I missed it. I have pushed the change.

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**